azure-proto-core/Resources/Plan.cs

```diff
@@ -17,12 +16,29 @@ public class Plan : IEquatable<Plan>, IComparable<Plan>
 
         public int CompareTo(Plan other)
         {
-            throw new NotImplementedException();
+            if (other == null) return 1;
+            if (this.Name == null && other.Name != null) return -1;
+            if (this.Name != null && this.Name.CompareTo(other.Name) != 0) return this.Name.CompareTo(other.Name);
```

Comment:
Same comment from Sku PR.  Store result of this.Name.CompareTo(other.Name) so you don't have to call it twice